internal/fields/validate.go

```diff
@@ -287,6 +309,21 @@ func (v *Validator) validateScalarElement(key string, val interface{}, doc commo
 	if definition == nil && skipValidationForField(key) {
 		return nil // generic field, let's skip validation for now
 	}
+
+	if definition == nil && !v.disabledImportAllECSSchema {
+		// import all fields from external schema
+		ecsSchema, err := v.FieldDependencyManager.ImportAllFields(defaultExternal)
+		if err != nil {
+			return err
+		}
```

Comment:
This will import ECS for each field, right?
Would it be possible to combine the `v.Schema` and the result of `v.FieldDependencyManager.ImportAllFields(defaultExternal)` in `CreateValidatorForDirectory(...)`? So we would only import ECS once and this logic here wouldn't be needed.

Reply:
I was worried in case all these ECS field definitions were added also for instance in the exported field documentation.

I've been checking again this, and when it is used to [get the exported fields](https://github.com/elastic/elastic-package/blob/main/internal/docs/exported_fields.go#L28):

```golang
	validator, err := fields.CreateValidatorForDirectory(fieldsParentDir)
	if err != nil {
		return "", errors.Wrapf(err, "can't create fields validator instance (path: %s)", fieldsParentDir)
	}
```

no spec version is used as a ValidatorOption parameter, so the default value (0.0.0) is taken. That makes sure that all these fields are not used at all in this stage since it must be at least 2.3.0.

I'll do the changes to move this logic directly in `CreateValidatorForDirectory()`

Thanks!